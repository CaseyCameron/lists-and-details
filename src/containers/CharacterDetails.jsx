import React, { useEffect, useState } from 'react';
import CharacterDetail from '../components/characters/CharacterDetail';
import { fetchCharacterById } from '../services/CharacterFetch';
import { useParams } from 'react-router-dom';

const ListAvatarCharacters = () => {
  const [loading, setLoading] = useState(true);
  const [characterDetail, setCharacterDetail] = useState([]);
  const { _id } = useParams();

  useEffect(async () => {
    const characterDetail = await fetchCharacterById(_id);
    setCharacterDetail(characterDetail);
    setLoading(false);
  }, [_id]);

  if (loading) return <h1>Loading...</h1>;
  return <CharacterDetail {...characterDetail} />
}

export default ListAvatarCharacters;