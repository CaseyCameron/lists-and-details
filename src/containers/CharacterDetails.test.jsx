import React from 'react';
import { render, screen } from '@testing-library/react';
import CharacterDetails from './CharacterDetails';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Character Detail Container', () => {
  it('Character Detail Page rendering test', async () => {
    render(
      <MemoryRouter initialEntries={['/5cf5679a915ecad153ab68d1']}>
        <Route exact path='/:_id'><CharacterDetails /></Route>
      </MemoryRouter>
    );

    screen.getByText('Loading...');

    const photoUrl = await screen.findAllByRole('img');
    expect(photoUrl).toHaveLength(1);
    expect(photoUrl).toMatchSnapshot();
  });
});