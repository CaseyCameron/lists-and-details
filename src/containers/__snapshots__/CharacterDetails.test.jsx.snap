// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Detail Container Character Detail Page rendering test 1`] = `
Array [
  <img
    alt="Analay"
    src="https://vignette.wikia.nocookie.net/avatar/images/c/c3/Analay.png/revision/latest?cb=20140430182201"
  />,
]
`;
